lib/puppet/provider/package/dnf.rb

```diff
@@ -30,6 +30,53 @@
 
   defaultfor :operatingsystem => :fedora, :operatingsystemmajrelease => ['22', '23']
 
+  def install
```

Comment:
So yum also allows the `upgrade` command for updating packages, so it might be easier to add this logic back up to the yum provider. Smaller footprint of changes, DRY-s up the code. Just need to see if `yum upgrade` is backwards compatible so it doesnt cause a regression for older releases...